src/Agent.Listener/Configuration/ConfigurationManager.cs

```diff
@@ -145,8 +147,10 @@ public AgentSettings LoadSettings()
                                                 args,
                                                 enforceSupplied);
                 Trace.Info("serverUrl: {0}", serverUrl);
+                Dictionary<string, string> credentialArgs = (args == null) ? new Dictionary<string, string>() : new Dictionary<string, string>(args);
+                credentialArgs[CliArgs.Url] = serverUrl;
 
-                credProv = AcquireCredentials(args, enforceSupplied);
+                credProv = AcquireCredentials(credentialArgs, enforceSupplied);
```

Comment:
I can't figure out, why this change? Consider adding a comment in the source code here if you think it's appropriate. I'm less familiar with this code, so perhaps it's actually something obvious and doesn't warrant a comment inside the source code itself.